mindmeld/components/question_answerer.py

```diff
@@ -451,803 +528,2178 @@ def _generate_mapping_data(embedder_model, embedding_fields):
+            SORT_ORDER_DESC = "desc"
+            SORT_DISTANCE = "distance"
+            SORT_TYPES = {SORT_ORDER_ASC, SORT_ORDER_DESC, SORT_DISTANCE}
+
+            if sort_type not in SORT_TYPES:
                 raise ValueError(
-                    "Invalid value for sort type '{}'".format(self.sort_type)
+                    "Invalid value for sort type '{}'".format(sort_type)
                 )
 
-            if self.field == "location" and self.sort_type != self.SORT_DISTANCE:
+            if self.data_type == "location" and sort_type != SORT_DISTANCE:
                 raise ValueError(
-                    "Invalid value for sort type '{}'".format(self.sort_type)
+                    "Invalid value for sort type '{}'".format(sort_type)
                 )
 
-            if self.field == "location" and not self.location:
+            if self.data_type == "location" and not location:
                 raise ValueError(
                     "No origin location specified for sorting by distance."
                 )
 
-            if self.sort_type == self.SORT_DISTANCE and self.field != "location":
+            if sort_type == SORT_DISTANCE and self.data_type != "location":
                 raise ValueError(
                     "Sort by distance is only supported using 'location' field."
                 )
 
-            # validate the sort field is number, date or location field
-            if not (
-                self.field_info.is_number_field()
-                or self.field_info.is_date_field()
-                or self.field_info.is_location_field()
-            ):
+            if self.data_type not in ["number", "date", "location"]:
                 raise ValueError(
                     "Custom sort criteria can only be defined for"
                     + " 'number', 'date' or 'location' fields."
                 )
+
+        def do_sort(self, curated_docs, sort_type, location=None):
+            # TODO: combining the sort-field values &  _scores is pretty naive, should be improved
+
+            if not curated_docs:
+                return curated_docs
+
+            self.do_sort_validation(sort_type, location)
+
+            if self.data_type == "number":
+                # obtain scores for this field values, normalize and multiply with _scores field
+                #    in docs and sort them
+                # _scores will all be zeros in case no similarity match is computed before sort
+                inds, common_ids, _scores = zip(
+                    *[(ii, doc["_id"], doc["_score"]) for ii, doc in enumerate(curated_docs) if
+                      doc["_id"] in self.id2value])
+                field_value_scores = [self.number_scorer(self.id2value[_id]) for _id in common_ids]
+
+                _scores = self.min_max_normalizer(_scores)
+                field_value_scores = self.min_max_normalizer(field_value_scores)
+                final_scores = [0.5 * a + 0.5 * b for a, b in zip(_scores, field_value_scores)]
```

Comment:
why do we do this?

Reply:
Edited, and we no more do such averaging.